### PR TITLE
CI: Enabled running tests on pull requests

### PR DESCRIPTION
**Pull request by stemann**
* CI: Updated test workflow from PkgTemplates generated CI workflow
  * Also, added julia-cache and codecov-action actions.
* Fixed (TIFF) tests on Windows

### CODE REVIEW

**Comment by stemann** [comment]
Might want to <strike>allow failure</strike>fix tests on Windows <strike>for the time being...</strike>.

**Comment by stemann** [comment]
The TIFF tests on Windows were misbehaving since the same file was opened after having been previously opened with mmap: https://github.com/tlnagy/TiffImages.jl/blob/v0.10.0/src/utils.jl#L139

Example of TIFF tests failing on Windows: https://github.com/JuliaIO/ImageIO.jl/actions/runs/8783191936/job/24098831147?pr=69#step:4:497

**Comment by stemann** [comment]
I guess https://github.com/JuliaIO/ImageIO.jl/actions/runs/8783191936 for #69 ran due to dependabot yielding a `push` event and not just a `pull_request` event, so adding
```yaml
on:
  pull_request:
```
to the CI/test workflow yaml is definitely needed.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          arch: ${{ matrix.arch }}
+      - uses: julia-actions/cache@v1
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: julia-actions/julia-runtest@v1
+        env:
+            JULIA_NUM_THREADS: 2
+      - uses: julia-actions/julia-processcoverage@v1
+      - uses: codecov/codecov-action@v3
+        with:
+          files: lcov.info
```
Comment (nhz2) [inline .github/workflows/CI.yml]:
```suggestion
      - uses: codecov/codecov-action@v4
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          files: lcov.info
```

**Comment by nhz2** [comment]
To fix the invalidations test, I think https://github.com/JuliaIO/ImageIO.jl/blob/deb2972a98e3e61a287686659499039c6d8a6b84/.github/workflows/InvalidationFlagger.yml#L10 needs to be changed to `version: '1'`

File: .github/workflows/CI.yml (hunk 1 above)
Comment (stemann) [reply .github/workflows/CI.yml]: Fixed

**Review by nhz2 — APPROVED** [review]
LGTM. Thank you for fixing this.

**Pull request merged**